airflow/configuration.py

```diff
@@ -1439,11 +1439,13 @@ def as_dict(
             ("default", self._default_values),
             ("airflow.cfg", self),
         ]
+        # Force the providers configuration to be loaded
+        self.configuration_description = retrieve_configuration_description()
```

Comment:
```suggestion
        if not self.configuration_description:
             self.configuration_description = retrieve_configuration_description()
```

perhaps?

Reply:
It’s there just that it doesn’t have providers configuration. so this would lead to old results. 
where it was set initially, it was set not to load providers configuration.

Reply:
Ah. Probably needs a comment then

Reply:
Or rather, expanding on the comment you did add :facepalm:

Reply:
That's not the right fix (it will cause much slowe airflow import if we do it this way because importing airflow is initializing configuraiton (this is why by default it does not load provider configuration).

We are lazy-loading provider's configuration by ProvidersManager.iniitialize_providers_configuration(). I have a fix coming with modified tests. I noticed that teh test that was retrieving config for non-sensitive values was not **really** testing if configuration for non-sensitive values was working - it checked only if as_dict() was called with include_sensitive=False (but it did not really check if sensitive values were replaced).

Reply:
https://github.com/apache/airflow/pull/38794